src/chesster.js

```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
+    if (!alternateOptions || !_.isEqual(channel.name, alternateOptions.channel)) {
+        return;
+    }
+    var heltourOptions = message.league.options.heltour;
+    if (!heltourOptions) {
+        winston.error("{} league doesn't have heltour options!?".format(message.league.options.name));
+        return;
+    }
+    
+    var components = message.text.split(" ");
+    var args = _.map(components.slice(0, 9), _.toLower);
+    var teamName = components.splice(9).join(" ");
+    var assign      = args[0],
+        player      = args[1],
+        to          = args[2],
+        board       = args[3],
+        boardNumber = args[4], 
+        during      = args[5], 
+        round       = args[6],
+        roundNumber = args[7],
+        on          = args[8];
+
+    var speaker = slack.getSlackUserFromNameOrID(message.user);
+    var speakerTeam = message.league.getTeamByPlayerName(speaker.name);
+    
+    if (!isCaptainOrModerator(speaker, speakerTeam, teamName)) {
+        replyOnlyACaptainOrModeratorCanDoThat(bot, message);
+        return;
+    }
+
+    // Ensure the basic command format is valid
+    if (!_.isEqual(["assign", "to", "board", "during", "round", "on"], [assign, to, board, during, round, on])) {
+        replyMisunderstoodAlternateAssignment(bot, message);
+        return;
+    }
+
+    boardNumber = parseInt(boardNumber, 10);
+    if(isNaN(boardNumber)){
+        replyMisunderstoodAlternateUnassignement(bot, message);
+        return;
+    }
+
+    roundNumber = parseInt(roundNumber, 10);
+    if(isNaN(roundNumber)){
+        replyMisunderstoodAlternatesUnassignment(bot, message);
+        return;
+    }
+
+    var team = message.league.getTeam(teamName);
+    if(!team){
+        replyUnrecognizedTeam(bot, message, teamName);
+        return;
+    }
+
+    heltour.assignAlternate(heltourOptions, roundNumber, team.number, boardNumber, player).then(function(){
+        bot.reply(message, player + " has been assigned to board " + boardNumber + " for " + teamName + " during round " + roundNumber);
+    }).catch(function(error){
+        bot.reply(message, "I failed to update your alternate assignement: " + error);
+        bot.reply(message, "Please contact a moderator.");
+    });
+});
+
+function replyMisunderstoodAlternateAssignment(bot, message){
+    bot.reply(message, "Sorry, I did not understand your alternate assignment");
+    bot.reply(message, "Please use the following format:");
+    bot.reply(message, "`assign <player> to board <board-number> during round <round-number> on <team-name>`");
+}
+
+/* alternate unassignment */
+/* unassign alternate for board <board-number> during round <round-number> on <team-name> */
+/* look up the original player and assign him to his board */
+chesster.hears({
+    middleware: [ slack.withLeagueByChannelName ],
+    patterns: [ '^unassign' ],
+    messageTypes: [ 
+        'ambient'
+    ]
+}, 
+function(bot, message){
+    var channel = channels.byId[message.channel];
```

Comment:
This whole preamble feels like it should be a middleware.  But that's not required for this PR.